src/gpg2john.c

```diff
@@ -128,22 +128,19 @@ private bz_stream bz;
 
 #define NULL_VER -1
 
-#define BIG_ENOUGH 0x10000
-
 /* Global Stuff */
 
-static unsigned char d[BIG_ENOUGH];
-static unsigned char u[BIG_ENOUGH];
-static unsigned char p[BIG_ENOUGH];
-static unsigned char q[BIG_ENOUGH];
-static unsigned char g[BIG_ENOUGH];
-static unsigned char y[BIG_ENOUGH];
-static unsigned char n[BIG_ENOUGH];
-static unsigned char e[BIG_ENOUGH];
-// static unsigned char x[BIG_ENOUGH];
-static unsigned char m_data[BIG_ENOUGH];
-static char gecos[BIG_ENOUGH];
-static char last_hash[2 * BIG_ENOUGH];
+static unsigned char d[16384]; // used for RSA, more than big enough
+static unsigned char u[16384]; // used for RSA, more than big enough
+static unsigned char p[16384];
+static unsigned char q[16384];
+static unsigned char g[16384];
+static unsigned char y[16384];
+static unsigned char n[16384];
+static unsigned char e[16384];
+static unsigned char m_data[90000];  // FIXME: I think 8192 is biggest data per block ???
+static char gecos[1024];
+static char *last_hash;
 static unsigned char m_salt[64];
```

Comment:
We may want to talk about how this is done.  I removed all 'BIG_ENOUGH' stuff.  We may still want to go that way, but it would be better to know max sizes.

The d/u/p/q... should have some form of max size (if 8192 bits max???).   Also, I have not seen any data blocks larger than 8192. That does not mean that larger blocks do not happen, but it appears gpg chops up data to that max size.